src/java/com/sarality/app/view/list/SearchListViewAdapter.java

```diff
@@ -15,16 +14,16 @@
  *
  * @author sunayna@dothat.in sunayna
  */
-public class SearchListComponentAdapter<T> extends ListComponentAdapter<T> implements Filterable {
```

Comment:
You don't need to do this. The ListViewInitializer already supports filtering and handles in internally. See if you need need classes at all now.